utils/anautilities.py

```diff
@@ -442,6 +442,62 @@ def filePathExists(searchPath, subPath=None, debug=False):
             print "Found %s"%s(testPath)
         return True
 
+# Find Inflection Point /////////////////////////////////////////////////////////////////
+#----------------------------------------------------------------------------------------
+
+def findInflectionPts(graph):
+    '''
+    Find and return the "knee", inflection point, of the SBit Rate scan
+    graph should be a TGraph or TH1F of the SBit Rate scan for 1 vfat 
+    '''
+
+
+    import ROOT as root
+    import numpy as np
+    from itertools import groupby
+
+    # make TH1F into TGraph
+    if type(graph) == root.TH1F :
+        graph = root.TGraph(graph)
+
+    # make arrays out of the x and y components
+    x = graph.GetX()
+    y = graph.GetY()
+    x = np.array(x, dtype=float)
+    y = np.array(y, dtype=float)
+
+    # Calculate the gradient of y as a function of x
+    # Documentation here: https://docs.scipy.org/doc/numpy/reference/generated/numpy.gradient.html
+    grad = np.gradient(y, x)
+
+    # Split the array when the slope changes sign and store the partitions of gradients
+    posGrad = [list(g) for k, g in groupby(grad, lambda x: x > 0) if k]
+    negGrad = [list(g) for k, g in groupby(grad, lambda x: x < 0) if k]
+   
+    # Sum the split arrays and find the most negative sum
+    negSum = []
+    bigNegSum = 0
+    for iNegArr in range(0, len(negGrad) ) :
+        tmpNegSum = sum(negGrad[iNegArr] )
+        negSum.append(tmpNegSum)
+        if tmpNegSum < bigNegSum :
+            bigNegSum = tmpNegSum
+            bigIdx =  iNegArr
+
+    # Get the inflection point
+    # We are defining the inflection point as the point where the most negative gradient sum begins
+    inflxGrad = negGrad[bigIdx][0]
+    try: 
+        inflxIdx = np.where(grad == inflxGrad) #return the index at the specified value
+    except ValueError:
+        print("Warning: the inflection point gradient (inflxGrad) value {:f} was not found in the gradient list".format(inflxGrad))
+    try:
+        inflxPnt = (x[inflxIdx], y[inflxIdx] )
+    except NameError:
+        print("Error: No Inflection Point was not found")
+
+    return inflxPnt
```

Comment:
this will still throw an exception since lines 494-497 don't actually resolve the `NameError`

Reply:
you can import `printRed` I believe from:

```python
from gempython.utils.gemlogger import printRed
```